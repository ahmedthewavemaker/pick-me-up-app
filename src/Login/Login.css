.forma{
    font-size: 25px;
    font-family: Cambria;
}
.button{
    margin:10px;
    background-color: rebeccapurple;
    font-size: 30px;
    color: gold;
    font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
    border-color: whitesmoke;
    border-radius: 10px;
    padding: 5px;
}
button:hover{
    color: lawngreen;
}
 