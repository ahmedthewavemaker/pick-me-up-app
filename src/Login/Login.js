import React,{useState} from 'react'
import AuthApiService from '../services/auth-service'
import './Login.css'

function Login(props) {

    const [error, setError] = useState(null)

   const handleSubmitJwtAuth = ev => {
        ev.preventDefault()
        setError(null)
        const { user_name, password } = ev.target
    
        AuthApiService.postLogin({
          user_name: user_name.value,
          password: password.value,
        })
          .then(res => {
            user_name.value = ''
            password.value = ''
            props.history.push('/create')
          })
          .catch(res => {
            console.log(res)
            setError( res.error )
          })
      }

    return (
        <div className='sign'>
          {error}
            <form onSubmit={handleSubmitJwtAuth} className='forma'>
                <label htmlFor='user_name'> Username: </label>
                <input placeholder='funUsername' type='text' name='user_name' id='user_name' />
                <br/>
                <label htmlFor='password'> Password: </label>
                <input placeholder='password' type='text' name='password' id='password' />
                <br/>
                <button className='button'> Submit </button>
                </form>
        </div>


    )
}

export default Login;