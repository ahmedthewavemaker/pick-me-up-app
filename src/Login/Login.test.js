import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Login from './Login';
import { MemoryRouter } from 'react-router-dom';

describe('Login component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter> <Login />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter> <Login  />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})