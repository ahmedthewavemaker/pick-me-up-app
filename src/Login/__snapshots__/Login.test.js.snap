// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="sign"
  >
    <form
      className="forma"
      onSubmit={[Function]}
    >
      <label
        htmlFor="user_name"
      >
         Username: 
      </label>
      <input
        id="user_name"
        name="user_name"
        placeholder="email@site.com"
        type="text"
      />
      <br />
      <label
        htmlFor="password"
      >
         Password: 
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        type="text"
      />
      <br />
      <button
        className="button"
      >
         Submit 
      </button>
    </form>
  </div>,
]
`;
